src/sentry/models/releasefile.py

```diff
@@ -171,34 +171,39 @@ def extract(self) -> TemporaryDirectory:
         return temp_dir
 
 
-def merge_release_archives(file1: IO, archive2: ReleaseArchive, target: IO):
+def merge_release_archives(file1: IO, archive2: ReleaseArchive, target: IO) -> bool:
     """Append contents of archive2 to copy of file1
 
     Skip files that are already present in archive 1.
-    """
-    # Create a copy
-    file1.seek(0)
-    target.write(file1.read())
 
+    :returns: True if zip archive was written to target
+    """
     with ReleaseArchive(file1) as archive1:
         manifest = archive1.manifest
 
-    files = manifest.get("files", {})
-    files2 = archive2.manifest.get("files", {})
+        files = manifest.get("files", {})
+        files2 = archive2.manifest.get("files", {})
+
+        if not files2.keys() - files.keys():
```

Comment:
_nit_: Since you already compute `files2 - files1` here, you can iterate over the result below. 

Another _nit_: We could log cases where a file is in both ZIPs but with different checksums. Our expectation would be that this does not happen often.